File: app/src/main/java/com/roman/kubik/spivanyklicejista/presentation/song/di/SongComponent.java
package com.roman.kubik.spivanyklicejista.presentation.song.di;

import com.roman.kubik.spivanyklicejista.general.di.ActivityScope;
import com.roman.kubik.spivanyklicejista.presentation.song.SongActivity;

import dagger.Subcomponent;

/**
 * Created by kubik on 1/20/18.
 */

@ActivityScope
@Subcomponent(modules = SongModule.class)
public interface SongComponent {

    void inject(SongActivity songActivity);

}

File: app/src/main/java/com/roman/kubik/spivanyklicejista/presentation/song/di/SongComponent.kt
package com.roman.kubik.spivanyklicejista.presentation.song.di

import com.roman.kubik.spivanyklicejista.general.di.ActivityScope
import com.roman.kubik.spivanyklicejista.presentation.song.SongActivity

import dagger.Subcomponent

/**
 * Created by kubik on 1/20/18.
 */

@ActivityScope
@Subcomponent(modules = arrayOf(SongModule::class))
interface SongComponent {

    fun inject(songActivity: SongActivity)

}
